gobblin-core/src/main/java/gobblin/source/extractor/extract/google/webmaster/GoogleWebMasterSource.java

```diff
@@ -0,0 +1,112 @@
+package gobblin.source.extractor.extract.google.webmaster;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.extract.QueryBasedSource;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class GoogleWebMasterSource extends QueryBasedSource<String, String[]> {
+
+  /**
+   * Provide the location for your Google Service Account private key. This can be generated in your Google API Manager.
+   */
+  public static final String KEY_CREDENTIAL_LOCATION = "source.google_webmasters.credential.location";
+  /**
+   * Provide the property site URL whose google search analytics data you want to download
+   */
+  public static final String KEY_PROPERTY = "source.google_webmasters.property";
+  /**
+   * This configuration defaults to false. Set it to true ONLY when you extremely care about the completeness of the available data set.
+   * However, if you set it to true, the code will throw an exception if it cannot achieve the perfect completeness of available data set.
+   * Set it to false if you can "endure" the incompleteness, which normally should be very minimal.
+   */
+  public static final String KEY_REQUEST_HYPERCRITICAL = "source.google_webmasters.request.isHypercritical";
+  /**
+   * Give a Google API service scope.
+   * For Webmaster, only two scopes are supported. WebmastersScopes.WEBMASTERS_READONLY and WebmastersScopes.WEBMASTERS
+   */
+  public static final String KEY_API_SCOPE = "source.google.api.scope";
+  /**
+   * The filters that will be passed to all your API requests.
+   * Filter format is [GoogleWebmasterFilter.Dimension].[DimensionValue]
```

Comment:
An example of the filter in the configuration file is Country.USA, where Country is GoogleWebmasterFilter.Dimension, USA is the value for the Country Dimension. Not quite sure what do you mean by GoogleWebmasterFilter.Dimension[DimensionValue]